tests/upgrade/customPassword_test.js

```diff
@@ -0,0 +1,79 @@
+const assert = require('assert');
+const { SERVICE_TYPE } = require('../helper/constants');
+
+Feature('PMM upgrade tests for custom password');
+const { dashboardPage } = inject();
+
+const clientDbServices = new DataTable(['serviceType', 'name', 'metric', 'annotationName', 'dashboard', 'upgrade_service']);
+
+clientDbServices.add([SERVICE_TYPE.MYSQL, 'ps-single', 'mysql_global_status_max_used_connections', 'annotation-for-mysql', dashboardPage.mysqlInstanceSummaryDashboard.url, 'mysql']);
+clientDbServices.add([SERVICE_TYPE.POSTGRESQL, 'pgsql_pgss_pmm', 'pg_stat_database_xact_rollback', 'annotation-for-postgres', dashboardPage.postgresqlInstanceSummaryDashboard.url, 'postgresql']);
+// clientDbServices.add([SERVICE_TYPE.MONGODB, 'rs101', 'mongodb_connections', 'annotation-for-mongo', dashboardPage.mongoDbInstanceSummaryDashboard.url, 'mongodb']);
+
+Data(clientDbServices).Scenario(
+  'Adding custom agent Password, Custom Label before upgrade At service Level @pre-custom-password-upgrade',
+  async ({
+    I, inventoryAPI, current,
+  }) => {
+    const {
+      serviceType, name, upgrade_service,
+    } = current;
+    const {
+      service_id, node_id, address, port,
+    } = await inventoryAPI.getServiceDetailsByPartialName(name);
+
+    const { agent_id: pmm_agent_id } = await inventoryAPI.apiGetPMMAgentInfoByServiceId(service_id);
+    let output;
+
+    switch (serviceType) {
+      case SERVICE_TYPE.MYSQL:
+        output = await I.verifyCommand(
+          `pmm-admin add mysql --node-id=${node_id} --pmm-agent-id=${pmm_agent_id} --port=${port} --password=GRgrO9301RuF --host=${address} --query-source=perfschema --agent-password=uitests --custom-labels="testing=upgrade" upgrade-${upgrade_service}`,
```

Comment:
Is it okay to have credentials in the commands? (password, agent-password, username (command below) etc.)

Reply:
Agent password is okay, its just for this test, other passwords added to the credentials file.